integration/tests/cook/test_basic.py

```
@@ -2626,6 +2626,72 @@ def test_pool_specific_quota_check_on_submit(self):
                 self.assertEqual(422, resp.status_code, msg=resp.content)
 
+
+    @unittest.skipUnless(util.pool_quota_test_pool() is not None, 'Test requires a test pool.')
+    def test_pool_quota(self):
+        settings = util.settings(self.cook_url)
+        quotas_map = settings.get("pools", {}).get("quotas", [])
+        pool_name = util.pool_quota_test_pool()
+        match_quota = [ii["quota"] for ii in quotas_map if
+                                   re.match(ii["pool-regex"], pool_name)]
+        # If there are no supported GPU models for pool, assert submission gets rejected
+        logging.info("Quota: " + repr(match_quota))
+        if(len(match_quota) == 0):
+            self.skipTest(f"Pool {pool_name} lacks quota assignment.")
+        quota = match_quota[0]
+        logging.info(f"Pool quota: {quota}")
+        if (quota["count"] >= 10):
+            self.skipTest(f"Job count quota too large for test")
+        quota_count = quota["count"]
+        job_count = quota_count + 2
+        job_mem = 16
+        job_cpus = .01
+
+        if quota["mem"] < job_mem * (job_count + 3):
+            self.skipTest("Quota memory too small for test")
+        if quota["cpus"] < job_cpus * (job_count + 3):
+            self.skipTest("Quota cpus to small for test")
+
+        # Now lookup the user quota and make sure it fits.
+        pools, resp = util.all_pools(self.cook_url)
+
+        user = self.determine_user()
+        resp = util.get_limit(self.cook_url, 'quota', user)
+        user_quota = resp.json()['pools'][pool_name]
+        logging.info(f"User quota {user_quota}")
+        if user_quota["count"] < job_count + 3:
+            self.skipTest("User quota count too small for test")
+        if user_quota["mem"] < job_mem * (job_count + 3):
+            self.skipTest("User quota memory too small for test")
+        if user_quota["cpus"] < job_cpus * (job_count + 3):
+            self.skipTest("User Quota cpus to small for test")
```

Should we fail instead of skip here?

---

Yeah. Ok.